File: java-micronaut/src/main/java/com/claude/bakeoff/controller/HealthController.java
package com.claude.bakeoff.controller;

import io.micronaut.http.HttpResponse;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

import java.util.Map;

@Controller
public class HealthController {
    
    @Get("/health")
    public HttpResponse<Map<String, Object>> health() {
        Map<String, Object> response = Map.of(
            "status", "OK",
            "service", "java-micronaut",
            "timestamp", System.currentTimeMillis()
        );
        
        return HttpResponse.ok(response)
                .header("Cache-Control", "public, max-age=60")
                .header("Surrogate-Control", "max-age=60");
    }
}

File: kotlin-micronaut/src/main/kotlin/com/claude/bakeoff/controller/HealthController.kt
package com.claude.bakeoff.controller

import io.micronaut.http.HttpResponse
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller
open class HealthController {
    
    @Get("/health")
    open fun health(): HttpResponse<Map<String, Any>> {
        val response = mapOf<String, Any>(
            "status" to "OK",
            "service" to "kotlin-micronaut", 
            "timestamp" to System.currentTimeMillis()
        )
        
        return HttpResponse.ok(response)
            .header("Cache-Control", "public, max-age=60")
            .header("Surrogate-Control", "max-age=60")
    }
}